apps/librelingo-server/build.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+DIRNAME=$(dirname "$0")
+DEFAULT_OUTPUT_DIR="$DIRNAME/bin"
+if [ -n "$1" ]; then
+    WEBSITE_DIR="$1"    
+else
+    echo "This script packs the LibreLingo website into a single executable file"
+    echo "Usage: build.sh WEBSITE-DIR [OUTPUT-DIR]"
+    echo "where:"
+    echo "  WEBSITE-DIR is the directory which contains the files for the LibreLingo website"
+    echo "  OUTPUT-DIR will contain the built executable binary files (optional; default: '$DEFAULT_OUTPUT_DIR')"
+    exit 1
+fi
+
+OUTPUT_DIR=${2:-"$DEFAULT_OUTPUT_DIR"}
+OUTPUT_DIR=$(readlink -f "$OUTPUT_DIR")
+
+# Setup temporary build directory
+BUILD_DIR=$(mktemp -d)
+function cleanup {
+    rm -rf "$BUILD_DIR"
+}
+trap cleanup EXIT
```

Comment:
What is the reason this needs to be cleaned up using trap instead of simply putting 

```
rm -rf "$BUILD_DIR
```

at the end of the script?

Reply:
Correct me if I'm wrong, but `set -eux` at line 32 should cause the script to fail if any of the build commands do not succeed. In this case, the cleanup at the end of the script would not be executed.